Sources/StreamChat/ChatClient.swift

```diff
@@ -593,7 +593,16 @@ extension _ChatClient: ConnectionStateDelegate {
                     clientUpdater.reloadUserIfNeeded(
                         userConnectionProvider: .closure { _, completion in
                             tokenProvider() { result in
-                                completion(result)
+                                switch result {
+                                case let .success(token):
+                                    if token.isExpired {
+                                        completion(.failure(ClientError.ExpiredToken()))
```

Comment:
There's no backoff and retry mechanism, right?

Reply:
This is a backoff mechanism, because we're inside the code that reobtains expired tokens

Reply:
Can you add a test to check if reconnection is scheduled when we try to connect with an expired token?

Reply:
There was such a test already, and I added another one that checks for the new case

Reply:
I can't find the test where we check if there's a reconnection scheduled after an expired token error

Reply:
The name's `test_client_webSocketIsDisconnected_becauseTokenExpired_callsReloadUserIfNeeded`